### ISSUE TITLE
[Bug]: Minimal permissions not working as intended

### ISSUE THREAD

**Issue by ndeet** [btcpayserver/btcpayserver#5344]
### What is your BTCPay version?

BTCPay 1.11.5-rc1 but also true for previous verison 1.11.4 and likely earlier

### How did you deploy BTCPay Server?

docker

### What happened?

**1.) no "view invoices" permission causes no store access at all**
if that permission is not set then the store won't be accessible at all, also if you have the "manage pull payments" permission

**2.) Sub-permissions won't work:**
E.g. if you have a serverwide role of "pull payment manager" that only has the permission of "create non-approved pull payments" and "archive pull payments" then the "Pull payments" menu entry won't show up at all. But it should?

**3.) Reporting menu item leads to 403**
When you have the "view invoices" permission the "Reporting" menu items shows up but leads to a 403 permission denied page

### How did you encounter this bug?

see above description

### Are you sure this is a bug report?

- [X] I confirm this is a bug report

**Comment by ndeet** [comment]
@Kukks sorry for the delay testing this. 

I added a role pull payment manager with permissions: createnonapprovedpp... and archivepp:
- correct sidemenu entry shows up
- correct permission to create an archive pp
- error?: on store dropdown user sees "create store" link and also store creation works
- error?: there is no way for the user to logout as the whole "accound" menu is missing

**Comment by dennisreimann** [comment]
> error?: on store dropdown user sees "create store" link and also store creation works

As discussed in the dev call this is related to people being able to create stores for themselves. We need to tackle this in a separate PR and also with a policy which can be defined for creating stores.

 > error?: there is no way for the user to logout as the whole "accound" menu is missing
 
I cannot reproduce this on master. For me the account menu shows up and the code also doesn't contain any constraints regarding roles/permissions for the account menu.


![grafik](https://github.com/btcpayserver/btcpayserver/assets/886/cc974891-0be2-42dc-8429-e6ad0e3704cf)

**Comment by ndeet** [comment]
Store creation
Yes I think we should do that, e.g. maybe even go so far that admins can configure how many stores can be created which would be helpful for 3rd party hosts. Maybe along with the super admin view for 1.13 we could do that too.

Account
I just tried to reproduce but couldn't, the Account link is here. Seems I made had some hallucinations. All good. 

So, case closed for now, will do more testing with permissions.
